McCorrections/src/LeptonWeights.cpp

```diff
@@ -185,35 +194,50 @@ double LeptonWeights::GetTriggerEfficiency(EventInfoBase& eventInfo, bool isData
+                        eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_IsoMu22_2016))
+                    && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_IsoMu22_eta2p1_2016))
+                    prescaled_weight = prescaled_weight_IsoMu22_eta2p1_2016;
+
+                if(period == Period::Run2016 &&
+                    !(eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled_2016) ||
+                        eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_IsoMu22_eta2p1_2016))
+                    && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_IsoMu22_2016))
+                    prescaled_weight = prescaled_weight_IsoMu22_2016;
+
+                if(period == Period::Run2017 && !eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled_2017)
+                        && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_muTau_2017))
+                    prescaled_weight = prescaled_weight_muTau_2017;
+
                 return prescaled_weight * std::min(muon_single_eff * (1 - tau_eff) + muon_cross_eff * tau_eff, 1.);
         }
         else{
-            static const std::vector<std::string> triggerPaths_unPrescaled = {
+            static const std::vector<std::string> triggerPaths_unPrescaled_2017 = {
                 "HLT_IsoMu27_v"};
             if(period == Period::Run2017
-                    && !eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled)
-                    && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_muTau))
-                prescaled_weight = prescaled_weight_muTau;
+                    && !eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled_2017)
+                    && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_muTau_2017))
+                prescaled_weight = prescaled_weight_muTau_2017;
             return prescaled_weight * muonSF.GetTriggerEff(eventInfo.GetLeg(1).GetMomentum(), isData);
         }
 
     }
 
     else if(channel == Channel::MuMu){
-        static const std::vector<std::string> triggerPaths_unPrescaled = {
+        static const std::vector<std::string> triggerPaths_unPrescaled_2017 = {
             "HLT_IsoMu27_v"};
         if(period == Period::Run2017
-                && !eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled)
-                && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_muTau))
-            prescaled_weight = prescaled_weight_muTau;
+                && !eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_unPrescaled_2017)
+                && eventInfo.GetTriggerResults().AnyAcceptAndMatch(triggerPaths_Prescaled_muTau_2017))
+            prescaled_weight = prescaled_weight_muTau_2017;
```

Comment:
add a comment for the future that for muMu for period Run2016 prescales should also be applied if all single mu triggers are used.